singularity/code/base.py

```diff
@@ -418,6 +428,11 @@ def sort_tuple(self):
         # bases aren't considered equal.
         return (-self.spec.size, -self.cpu, self.name, id(self))
 
+    def finish(self, is_player=True, loading_savegame=False):
+        super(Base, self).finish(is_player=is_player, loading_savegame=loading_savegame)
+        self.recalc_cpu()
+        self.check_power()
```

Comment:
This looks like it might give issues during loading a save game.  The `check_power()` method calls `g.pl.recalc_cpu()` but during a save game, not all bases are restored at this point in time.  This makes the player object assume it hardly has any CPU and then drop CPU assignments accordingly.

Reply:
Also, the order of these calls look wrong; What is the point of doing a `Base.recalc_cpu()` before fixing the power value?  Even more so when `check_power()` does not fix the CPU, so recalc_cpu may have been done on the wrong power state leading to incorrect results.

I think there is a logical error here.

Reply:
Have looked at the code in more detail and the issue I described in my first comment to this bit is probably not an issue after all.  Accordingly, I have hidden that comment.